Description: Rename libslab to libmate-slab
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Forwarded: https://github.com/mate-desktop/mate-control-center/issues/100
Abstract:
 This is to avoid package name and version conflicts with
 libslab0 / libslab0a which existed in Debian until squeeze.
 .
 MATE components (currently only gnome-main-menu) depending
 on libmate-slab0 have to be patch accordingly. 
--- a/libslab/Makefile.am
+++ b/libslab/Makefile.am
@@ -1,5 +1,5 @@
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libslab.pc
+pkgconfig_DATA = mate-slab.pc
 
 AM_CPPFLAGS =					\
 	-I$(top_srcdir)				\
@@ -28,9 +28,9 @@
 	system-tile.h				\
 	tile.h
 
-lib_LTLIBRARIES = libslab.la
+lib_LTLIBRARIES = libmate-slab.la
 
-libslab_la_SOURCES =				\
+libmate_slab_la_SOURCES =			\
 	$(MARSHAL_GENERATED)			\
 	app-resizer.c				\
 	app-shell.c				\
@@ -55,9 +55,9 @@
 	tile-action.c				\
 	tile.c
 
-libslab_includedir = $(includedir)/libslab
-libslab_include_HEADERS = $(HEADER_FILES)
-libslab_la_LIBADD = $(LIBSLAB_LIBS)
+libmate_slab_includedir = $(includedir)/libmate-slab
+libmate_slab_include_HEADERS = $(HEADER_FILES)
+libmate_slab_la_LIBADD = $(LIBSLAB_LIBS)
 
 search-entry-watermark.h: search-entry-watermark.svg
 	echo '#define SEARCH_ENTRY_WATERMARK_SVG "\' > $@; \
--- a/configure.ac
+++ b/configure.ac
@@ -182,20 +182,20 @@
 PKG_CHECK_MODULES(GLIB, glib-2.0 $GMODULE_ADD)
 
 dnl
-dnl Check dependencies of libslab
+dnl Check dependencies of libmate-slab
 dnl
 PKG_CHECK_MODULES(LIBSLAB, [
 			   gio-2.0 mate-desktop-2.0 librsvg-2.0 libmate-menu pango
 			   ],
-			   have_libslab=yes,
-			   have_libslab=no)
+			   have_libmateslab=yes,
+			   have_libmateslab=no)
 WARN_CFLAGS="-Wall"
 
-AC_SUBST(LIBSLAB_CFLAGS)
-AC_SUBST(LIBSLAB_LIBS)
+AC_SUBST(LIBMATESLAB_CFLAGS)
+AC_SUBST(LIBMATESLAB_LIBS)
 AC_SUBST(WARN_CFLAGS)
 
-AM_CONDITIONAL(HAVE_LIBSLAB, [test $have_libslab = yes])
+AM_CONDITIONAL(HAVE_LIBMATESLAB, [test $have_libmateslab = yes])
 
 dnl
 dnl Check for Xft version 2; we build in extra functionality to the font capplet
@@ -371,7 +371,7 @@
 libwindow-settings/Makefile
 libwindow-settings/mate-window-settings-2.0.pc
 po/Makefile.in
-libslab/libslab.pc
+libslab/mate-slab.pc
 libslab/Makefile
 shell/Makefile
 shell/matecc.desktop.in
@@ -390,7 +390,7 @@
 
         GTK+ API version:      ${GTK_API_VERSION}
         Appindicator:          ${enable_appindicator}
-        Libslab:               ${have_libslab}
+        Libmate-slab:          ${have_libmateslab}
         XCursor support:       ${have_xcursor}
         XInput support:        ${have_xinput}
         Ubuntu hacks:          ${enable_ubuntu}
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,11 +1,11 @@
 SUBDIRS = po libwindow-settings capplets font-viewer help
 DIST_SUBDIRS = po libwindow-settings capplets font-viewer help libslab shell typing-break
 
-if HAVE_LIBSLAB
+if HAVE_LIBMATESLAB
 SUBDIRS += libslab
 endif
 
-if HAVE_LIBSLAB
+if HAVE_LIBMATESLAB
 SUBDIRS += shell
 endif
 
--- a/shell/Makefile.am
+++ b/shell/Makefile.am
@@ -1,5 +1,5 @@
 REAL_LIBSLAB_CFLAGS = -I$(top_srcdir)/libslab
-REAL_LIBSLAB_LIBS = $(top_builddir)/libslab/libslab.la
+REAL_LIBSLAB_LIBS = $(top_builddir)/libslab/libmate-slab.la
 
 AM_CPPFLAGS =					\
 	-I$(top_srcdir)				\
--- a/libslab/app-resizer.h
+++ b/libslab/app-resizer.h
@@ -24,7 +24,7 @@
 #include <glib.h>
 #include <gtk/gtk.h>
 
-#include <libslab/app-shell.h>
+#include "app-shell.h"
 
 #ifdef __cplusplus
 extern "C" {
--- a/libslab/app-shell.h
+++ b/libslab/app-shell.h
@@ -28,8 +28,8 @@
 #include <matemenu-tree.h>
 #include <libmate-desktop/mate-desktop-item.h>
 
-#include <libslab/slab-section.h>
-#include <libslab/tile.h>
+#include "slab-section.h"
+#include "tile.h"
 
 #ifdef __cplusplus
 extern "C" {
--- a/libslab/application-tile.h
+++ b/libslab/application-tile.h
@@ -21,7 +21,7 @@
 #ifndef __APPLICATION_TILE_H__
 #define __APPLICATION_TILE_H__
 
-#include <libslab/nameplate-tile.h>
+#include "nameplate-tile.h"
 
 #include <libmate-desktop/mate-desktop-item.h>
 
--- a/libslab/directory-tile.h
+++ b/libslab/directory-tile.h
@@ -23,7 +23,7 @@
 
 #include <time.h>
 
-#include <libslab/nameplate-tile.h>
+#include "nameplate-tile.h"
 
 #ifdef __cplusplus
 extern "C" {
--- a/libslab/nameplate-tile.h
+++ b/libslab/nameplate-tile.h
@@ -21,7 +21,7 @@
 #ifndef __NAMEPLATE_TILE_H__
 #define __NAMEPLATE_TILE_H__
 
-#include <libslab/tile.h>
+#include "tile.h"
 
 #include <gtk/gtk.h>
 
--- a/libslab/shell-window.h
+++ b/libslab/shell-window.h
@@ -25,7 +25,7 @@
 #include <gtk/gtk.h>
 #include <libmate-desktop/mate-desktop-item.h>
 
-#include <libslab/app-shell.h>
+#include "app-shell.h"
 
 #ifdef __cplusplus
 extern "C" {
--- a/libslab/slab.h
+++ b/libslab/slab.h
@@ -22,24 +22,24 @@
 #ifndef __SLAB_H__
 #define __SLAB_H__
 
-#include <libslab/app-resizer.h>
-#include <libslab/app-shell.h>
-#include <libslab/application-tile.h>
-#include <libslab/bookmark-agent.h>
-#include <libslab/directory-tile.h>
-#include <libslab/document-tile.h>
-#include <libslab/double-click-detector.h>
-#include <libslab/mate-utils.h>
-#include <libslab/libslab-utils.h>
-#include <libslab/nameplate-tile.h>
-#include <libslab/search-bar.h>
-#include <libslab/search-context-picker.h>
-#include <libslab/search-entry.h>
-#include <libslab/shell-window.h>
-#include <libslab/slab-mate-util.h>
-#include <libslab/slab-section.h>
-#include <libslab/system-tile.h>
-#include <libslab/tile.h>
+#include "app-resizer.h"
+#include "app-shell.h"
+#include "application-tile.h"
+#include "bookmark-agent.h"
+#include "directory-tile.h"
+#include "document-tile.h"
+#include "double-click-detector.h"
+#include "mate-utils.h"
+#include "libslab-utils.h"
+#include "nameplate-tile.h"
+#include "search-bar.h"
+#include "search-context-picker.h"
+#include "search-entry.h"
+#include "shell-window.h"
+#include "slab-mate-util.h"
+#include "slab-section.h"
+#include "system-tile.h"
+#include "tile.h"
 
 #endif /* __SLAB_H__ */
 
--- a/libslab/system-tile.h
+++ b/libslab/system-tile.h
@@ -21,7 +21,7 @@
 #ifndef __SYSTEM_TILE_H__
 #define __SYSTEM_TILE_H__
 
-#include <libslab/nameplate-tile.h>
+#include "nameplate-tile.h"
 
 #include <libmate-desktop/mate-desktop-item.h>
 
